File: packages/sdk-python/src/forgemsg/resources/events.py
from __future__ import annotations

from typing import Any, Dict, Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from ..client import ForgemsgClient


class EventsResource:
    def __init__(self, client: "ForgemsgClient") -> None:
        self._client = client

    def track(
        self,
        event_name: str,
        contact_id: Optional[str] = None,
        contact_email: Optional[str] = None,
        properties: Optional[Dict[str, Any]] = None,
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {"eventName": event_name}
        if contact_id:
            body["contactId"] = contact_id
        if contact_email:
            body["contactEmail"] = contact_email
        if properties:
            body["properties"] = properties
        return self._client.post("/api/v1/events", json=body)

    def list(
        self,
        contact_id: Optional[str] = None,
        limit: int = 50,
    ) -> Dict[str, Any]:
        params: Dict[str, Any] = {"limit": limit}
        if contact_id:
            params["contactId"] = contact_id
        return self._client.get("/api/v1/events", params=params)

File: packages/sdk/src/resources/events.ts
import type { ForgemsgClient } from '../client.js';
import type { CustomEvent, ItemResponse } from '../types.js';

export class EventsResource {
  constructor(private readonly client: ForgemsgClient) {}

  /** Track a custom event for a contact (triggers api_event workflows). */
  track(params: {
    contactId?: string;
    contactEmail?: string;
    eventName: string;
    properties?: Record<string, unknown>;
  }): Promise<ItemResponse<CustomEvent>> {
    return this.client.post<ItemResponse<CustomEvent>>('/api/v1/events', params);
  }

  list(params?: { contactId?: string; limit?: number }): Promise<{ data: CustomEvent[] }> {
    const qs = new URLSearchParams();
    if (params?.contactId) qs.set('contactId', params.contactId);
    if (params?.limit) qs.set('limit', String(params.limit));
    const q = qs.toString();
    return this.client.get(`/api/v1/events${q ? `?${q}` : ''}`);
  }
}
